tests/cache-cmd.earth

```diff
@@ -0,0 +1,32 @@
+VERSION 0.6
+
+FROM alpine:3.13
+CACHE --persist /base/persistent
+CACHE /base/ephemeral
+
+test-prev-prev:
+    WORKDIR /test
+    CACHE ./my/data
+    RUN echo "nothing to see here" >> ./my/data/hello
+    RUN echo "...or here" >> /base/ephemeral/hello
+    RUN echo "hello" >> /base/persistent/hello
+
+test-prev:
+    FROM +test-prev-prev
+    CACHE --persist ./my/data
+    CACHE --persist ./my/other
+    RUN echo "hello world" >> ./my/data/hello
+    RUN echo "hello again" >> ./my/other/hello
+    RUN echo "$(cat /base/persistent/hello), hello" > /base/persistent/hello
+
+test:
+    FROM +test-prev
+    # These files should contain only the contents written to it in +test-prev.
+    RUN test "$(cat ./my/data/hello)" = "hello world"
```

Comment:
will this still work when `./my/data/hello` contains multiple lines? should we be using `tail -n 1` instead?

Reply:
In a similar vein, can we write a test that validates that the cache contains items written from a previous run of earthly?
e.g.

Consider earth1.test
```
foo:
    CACHE ./my/data
    RUN echo alpha >> ./my/data/hello
```

Consider earth2.test
```
foo:
    CACHE ./my/data
    RUN echo bravo >> ./my/data/hello
```

then run `cp earth1.test Earthfile; earthly +foo`, followed by `cp earth2.test Earthfile; earthly +foo`.

Can we validate `./my/data/hello` ends with `alpha\nbravo\n`?

Reply:
Yes will look into that. Seems like a good test since that's an important part of the feature

Reply:
I attempted this test but it seems when we run the nested Earthfiles in the `tests/` director, they don't utilize cache (as if they're run with a `--no-cache` flag, but I don't see that being done explicitly anyway)